op-batcher/batcher/driver.go

```diff
@@ -382,6 +385,34 @@ const (
 	TxpoolCancelPending
 )
 
+// sendToThrottlingLoop sends the current pending bytes to the throttling loop.
+// It is not blocking, no signal will be sent if the channel is full.
+func (l *BatchSubmitter) sendToThrottlingLoop(pendingBytesUpdated chan int64) {
+	if l.Config.ThrottleThreshold == 0 {
+		return
+	}
+
+	l.channelMgrMutex.Lock()
+	pendingBytes := l.channelMgr.PendingDABytes()
+	l.channelMgrMutex.Unlock()
+
+	// notify the throttling loop it may be time to initiate throttling without blocking
+	select {
+	case pendingBytesUpdated <- pendingBytes:
+	default:
+	}
+}
+
+// signalPublishingLoop sends the current pending bytes to the throttling loop.
+// It is not blocking, no signal will be sent if the channel is full.
+func (l *BatchSubmitter) signalPublishingLoop(blocksLoaded chan struct{}) {
+	// notify the publishingLoop in a non blocking way
+	select {
+	case blocksLoaded <- struct{}{}:
+	default:
+	}
+}
```

Comment:
Note that this doesn't need to be a method on the `BatchSubmitter`, you don't use any of its state ;) could just be a generic `trySignal` function.